torch/csrc/jit/codegen/cuda/lower_predicate_elimination.cpp

```diff
@@ -294,6 +295,47 @@ class PredicateChcker : public IterVisitor {
       return true;
     }
 
+    // TODO: This is directed WAR on FusionPersistentNormLocalShared.
+    //  This use case along with other previous issues motivate a
+    //   joint optimization of predicate removal and buffer reuse.
+    // In this particular case:
+    //   __shared__ T0 [10], T1[10]
+    //   for i in ...
+    //      T1[i] = T0[i] + ...  // exp0
+    //      T2 = 0;              // init for exp1
+    //      if(pred)
+    //        T2 = T1 ...        // exp1
+    //  If we remove pred around expr1, as the way the pred removal
+    //    pass is set up, the init for expr will be pushed up to
+    //    initialize T1 instead.
+    //  However if we initialize T1, the code will look like:
+    //  for i in ...
+    //    T1[i] = 0;
+    //  for i in ...
+    //    if(pred)
+    //      T1[i] = T0[i] + ...
+    //  Note that we'd be able to reuse buffer of T0 for T1 but
+    //    if we initialze T1 we cannot do that and thus the
+    //    kernel would not fit in smaller devices.
```

Comment:
I think it's fine to disable in this case for now, but looks like for this kind of patterns we would want to just initialize `T0` and eliminate the predicates for `T1` and `T2`, which would not interfere with the buffer reuse between `T0` and `T1`.

Reply:
Yes. The holistic predicate/initialization analysis has been on the TODO list and quite a few TODO's in this PR actually boil down to that. I will try to get to that in follow ups.